### ISSUE TITLE
Question: Can I call a scheduled method manually with ShedLock support?

### ISSUE THREAD

**Issue by Somebodyisnobody** [lukas-krecan/ShedLock#2319]
Hello,
I didn't found other ways to get in touch and I also don't want to write somebody directly as I would like to share the information from the answer with other fellows.

**My case:**
I want to do large magic operations (some batch database stuff and generating items while communicating with other systems) in a Spring Boot application. For this I wrote a function which runs in a separate thread.
For automatic resumption in case the process dies I also want to run this function in a separate thread at the start of each instance of the application (Spring scheduled). To prevent parallel executions I want to use ShedLock.

So the batch function can be called manually (also by multiple instances as the trigger is an HTTP API) and can be called by Spring scheduling.

Is this possible to use ShedLock here for both "entrypoints" or is ShedLock only reliable when called by the Spring scheduler?
The question why I am asking is the following picture from the documentation. I don't know if the "ScheduledMethod AOP proxy" also runs reliable when calling the method manually.
https://github.com/lukas-krecan/ShedLock/blob/master/documentation/method_proxy.png

**Comment by vinz486** [comment]
I also have the same issue. I need to trigger routines manually: names and timing are dynamic.

**Comment by lukas-krecan** [comment]
Yes, PROXY_METHOD uses Spring AOP so the locking is applied even when called directly.

**Comment by Somebodyisnobody** [comment]
Thank you for the information!

**Comment by vinz486** [comment]
I'm using the "no framework way": https://github.com/lukas-krecan/ShedLock?tab=readme-ov-file#locking-without-a-framework

But, my concern is: the LockingTaskExecutor executor = new DefaultLockingTaskExecutor(lockProvider);
statement can be a singleton field? In other words, is thread safe?

Thank you.
